packages/mui-material/src/FormControlLabel/FormControlLabel.d.ts

```diff
@@ -45,7 +45,7 @@ export interface FormControlLabelProps
   /**
    * A text or an element to be used in an enclosing label element.
    */
-  label: string | number | React.ReactElement;
+  label: React.ReactNode;
```

Comment:
Going in the opposite direction of #29324 sounds great, but we still need to account for when the label is empty. We forgot about this in #25883, which lead to this bug: #29301. cc @michaldudak.

For the solution, I would propose 1. adding a test case for an empty label and 2. checking how `disableTypography` is implemented in the other components, and copying it (done better)

Reply:
Updated to match implementation of other components. Additionally, since there now is a nullish check I've set the label type to `label?: React.ReactNode`. I'm not entirely sure if it makes sense to allow dropping the property, might be some special case. But there's definitely a valid case to allow `null` and `undefined` in relation to translation frameworks, be it not displaying untranslated values or even asynchronously fetching translations.

Reply:
@michaldudak yeah. I've written a quick codesandbox for usage with `i18next` and `react-i18next` and you can see that without any additional typing it doesn't accept the result because it's possibly undefined.

https://codesandbox.io/s/react-typescript-forked-t10648?file=/src/App.tsx

![image](https://user-images.githubusercontent.com/9079379/158214357-7e86af4d-92c9-4456-87e5-2fd3875ad563.png)

Reply:
Damn I forgot that `React.ReactNode` includes `| undefined` so yeah, PR has been updated to make the prop required.